source/service.go

```diff
@@ -481,13 +482,18 @@ func (sc *serviceSource) generateEndpoints(svc *v1.Service, hostname string, pro
 			endpoints = append(endpoints, sc.extractHeadlessEndpoints(svc, hostname, ttl)...)
 		}
 	case v1.ServiceTypeNodePort:
-		// add the nodeTargets and extract an SRV endpoint
-		targets, err = sc.extractNodePortTargets(svc)
-		if err != nil {
-			log.Errorf("Unable to extract targets from service %s/%s error: %v", svc.Namespace, svc.Name, err)
-			return endpoints
+		// Is the service associated with aws load balancer controller v2?
+		if _, ok := svc.ObjectMeta.Annotations[awsLoadBalancerTypeAnnotation]; ok {
```

Comment:
Please have a look https://github.com/kubernetes-sigs/external-dns/blob/master/source/source.go#L134. This should help you. I'd like to move the code a bit

Reply:
Ah thank you @njuettner. I'll move things around now that I see this section.

Reply:
For posterity, [here is the permalink version of the source.go](https://github.com/kubernetes-sigs/external-dns/blob/979c2d611bb7a626dfaaade5fe36f7693949a733/source/source.go#L134) that was linked at the time of this original review.

@njuettner I have made an attempt at using the `getProviderSpecificAnnotations()` to fetch this particular annotation. However, this feels a bit off because most of the provider specific annotations are for use in external-dns providers, but this particular annotation is not. It is for use in the source, it has no impact to the external-dns provider at all. The purpose of checking this annotation is to detect when a NodePort service is being used by the aws load balancer controller as if it is a load balancer, because then the service source must be processed as a LoadBalancer instead of as a NodePort. The [aws-load-balancer-controller-v2](https://github.com/kubernetes-sigs/aws-load-balancer-controller/) actually creates Load Balancers in reaction to NodePort services with this annotation.